nebula_decoders/include/nebula_decoders/nebula_decoders_robosense/decoders/robosense_packet.hpp

```diff
@@ -0,0 +1,268 @@
+      channel_correction.elevation = corrected_vertical_angle.angles[i].getAngle();
+      calibration.calibration.push_back(channel_correction);
+    }
+    return calibration;
+  }
+};
+
+struct FirmwareVersion
+{
+  big_uint8_buf_t first_octet;
+  big_uint8_buf_t second_octet;
+  big_uint8_buf_t third_octet;
+  big_uint8_buf_t fourth_octet;
+  big_uint8_buf_t fifth_octet;
+
+  [[nodiscard]] std::string to_string() const
+  {
+    std::stringstream ss;
+    ss << std::hex << std::setfill('0') << std::setw(2) << static_cast<int>(first_octet.value())
+       << std::setw(2) << static_cast<int>(second_octet.value()) << std::setw(2)
+       << static_cast<int>(third_octet.value()) << std::setw(2)
+       << static_cast<int>(fourth_octet.value()) << std::setw(2)
+       << static_cast<int>(fifth_octet.value());
+    return ss.str();
+  }
+};
+
+struct SerialNumber
+{
+  big_uint8_buf_t first_octet;
+  big_uint8_buf_t second_octet;
+  big_uint8_buf_t third_octet;
+  big_uint8_buf_t fourth_octet;
+  big_uint8_buf_t fifth_octet;
+  big_uint8_buf_t sixth_octet;
+
+  [[nodiscard]] std::string to_string() const
+  {
+    std::stringstream ss;
+    ss << std::hex << std::setfill('0') << std::setw(2) << static_cast<int>(first_octet.value())
+       << std::setw(2) << static_cast<int>(second_octet.value()) << std::setw(2)
+       << static_cast<int>(third_octet.value()) << std::setw(2)
+       << static_cast<int>(fourth_octet.value()) << std::setw(2)
+       << static_cast<int>(fifth_octet.value()) << std::setw(2)
+       << static_cast<int>(sixth_octet.value());
+    return ss.str();
+  }
+};
+
+#pragma pack(pop)
+
+/// @brief Get the number of returns for a given return mode
+/// @param return_mode The return mode
+/// @return The number of returns
+size_t get_n_returns(ReturnMode return_mode)
+{
+  if (return_mode == ReturnMode::DUAL) {
+    return 2;
+  }
+  return 1;
+}
+
+/// @brief Get timestamp from packet in nanoseconds
+/// @tparam PacketT The packet type
+/// @param packet The packet to get the timestamp from
+/// @return The timestamp in nanoseconds
+template <typename PacketT>
+uint64_t get_timestamp_ns(const PacketT & packet)
+{
+  return packet.header.timestamp.get_time_in_ns();
+}
+
+/// @brief Get the distance unit of the given packet type in meters. Distance values in the packet,
+/// multiplied by this value, yield the distance in meters.
+/// @tparam PacketT The packet type
+/// @param packet The packet to get the distance unit from
+/// @return The distance unit in meters
+template <typename PacketT>
+double get_dis_unit(const PacketT & packet)
+{
+  // Packets define distance unit in millimeters, convert to meters here
+  const uint8_t range_resolution = packet.header.range_resolution.value();
+  if (range_resolution == 0) {
+    return 0.0050;
+  } else if (range_resolution == 1) {
+    return 0.0025;
+  }
+  // Should throw an error here?
+  // throw std::runtime_error("Unknown range resolution");
```

Comment:
Yes, please :smiley: That would get rid of the "control reaches end of non-void function" warning.

```suggestion
  throw std::runtime_error("Unknown range resolution");
```